src/PowerShellEditorServices/Services/PowerShell/Console/ConsoleReadLine.cs

```diff
@@ -5,52 +5,46 @@
 
 using System.Text;
 using System.Threading;
-using System.Threading.Tasks;
 
 namespace Microsoft.PowerShell.EditorServices.Services.PowerShell.Console
 {
     using Microsoft.PowerShell.EditorServices.Services.PowerShell.Execution;
     using Microsoft.PowerShell.EditorServices.Services.PowerShell.Host;
     using System;
     using System.Collections.Generic;
+    using System.Diagnostics;
     using System.Management.Automation;
     using System.Management.Automation.Language;
-    using System.Runtime.CompilerServices;
     using System.Security;
 
     internal class ConsoleReadLine : IReadLine
     {
         private readonly PSReadLineProxy _psrlProxy;
 
-        private readonly EditorServicesConsolePSHost _psesHost;
-
-        private readonly PowerShellExecutionService _executionService;
+        private readonly InternalHost _psesHost;
 
         private readonly EngineIntrinsics _engineIntrinsics;
 
         #region Constructors
 
         public ConsoleReadLine(
             PSReadLineProxy psrlProxy,
-            EditorServicesConsolePSHost psesHost,
-            PowerShellExecutionService executionService,
+            InternalHost psesHost,
             EngineIntrinsics engineIntrinsics)
         {
             _psrlProxy = psrlProxy;
             _psesHost = psesHost;
-            _executionService = executionService;
             _engineIntrinsics = engineIntrinsics;
         }
 
         #endregion
 
         #region Public Methods
 
-        public Task<string> ReadLineAsync(CancellationToken cancellationToken) => ReadLineAsync(isCommandLine: true, cancellationToken);
-
-        public string ReadLine() => ReadLineAsync(CancellationToken.None).GetAwaiter().GetResult();
-
-        public SecureString ReadSecureLine() => ReadSecureLineAsync(CancellationToken.None).GetAwaiter().GetResult();
+        public string ReadLine(CancellationToken cancellationToken)
```

Comment:
The big change in this file is that there's no more async implementation. Since we're always calling this from our dedicated pipeline thread, synchronous is what we want. That also maps better to the underlying implementation. We're not blocking a threadpool worker just to call readline any more.